src/codegen.c

```diff
@@ -139,8 +141,10 @@ static LLVMValueRef codegen_constant(const struct State *st, const Constant *c)
         return LLVMConstInt(LLVMInt1Type(), c->data.boolean, false);
     case CONSTANT_INTEGER:
         return LLVMConstInt(codegen_type(type_of_constant(c)), c->data.integer.value, c->data.integer.is_signed);
+    case CONSTANT_FLOAT:
+        return LLVMConstRealOfString(codegen_type(type_of_constant(c)), c->data.double_or_float_text);
     case CONSTANT_DOUBLE:
-        return LLVMConstRealOfString(codegen_type(type_of_constant(c)), c->data.double_text);
+        return LLVMConstRealOfString(codegen_type(type_of_constant(c)), c->data.double_or_float_text);
```

Comment:
okay

Reply:
but how to merge?

Reply:
```c
case CONSTANT_FLOAT || CONSTANT_DOUBLE:
       return LLVMConstRealOfString(codegen_type(type_of_constant(c)), c->data.double_or_float_text);
```
syntax doesn't seem right

Reply:
goto looks good:
```c
    case CONSTANT_FLOAT:
        constantfloat:
        return LLVMConstRealOfString(codegen_type(type_of_constant(c)), c->data.double_or_float_text);
    case CONSTANT_DOUBLE:
        goto constantfloat;
```

Reply:
but, it will strangely made return 0 into an error:
![image](https://user-images.githubusercontent.com/71159641/216027478-ce277f44-5c30-4b7e-8a28-861ed8d9fbd6.png)
test:
```python
declare printf(pattern: byte*, ...) -> int

def test(x: float) -> float:
    return x

def main() -> int:
    printf("%f", test(1.1))
    return 0
```
return
```
jou.exe argfloat.jou
compiler error in file "argfloat.jou", line 8: invalid number or variable name "0"
The process failed with status 1.
```
srouce code:
```c
static LLVMValueRef codegen_constant(const struct State *st, const Constant *c)
{
    switch(c->kind) {
    case CONSTANT_BOOL:
        return LLVMConstInt(LLVMInt1Type(), c->data.boolean, false);
    case CONSTANT_INTEGER:
        return LLVMConstInt(codegen_type(type_of_constant(c)), c->data.integer.value, c->data.integer.is_signed);
    case CONSTANT_FLOAT:
        constantfloat:
        return LLVMConstRealOfString(codegen_type(type_of_constant(c)), c->data.double_or_float_text);
    case CONSTANT_DOUBLE:
        goto constantfloat;
    case CONSTANT_NULL:
        return LLVMConstNull(codegen_type(voidPtrType));
    case CONSTANT_STRING:
        return make_a_string_constant(st, c->data.str);
    }
    assert(0);
}
```

Reply:
fixed in the latest commit